### PR TITLE
Don't validate edges multiple times

### PR DESCRIPTION
**Pull request by pmnoxx**
We are validating edges multiple times, even if we already have them in our routing table.

### CODE REVIEW

File: chain/network/src/peer_manager.rs [hunk 1]
```diff
@@ -1450,12 +1450,24 @@ impl Handler<NetworkRequests> for PeerManagerActor {
...
+                    .filter(|edge| {
+                        if let Some(cur_edge) =
+                            self.routing_table.get_edge(edge.peer0.clone(), edge.peer1.clone())
+                        {
+                            // only consider edges with bigger nonce
+                            return cur_edge.nonce < edge.nonce;
+                        }
+                        true
+                    })
+                    .collect();
```
Comment (matklad) [inline chain/network/src/peer_manager.rs:1464]:
```suggestion
                let SyncData { mut edges, accounts } = sync_data;
                edges.retain(|edge| {
                    match self.routing_table.get_edge(edge.peer0.clone(), edge.peer1.clone()) {
                        // only consider edges with bigger nonce
                        Some(cur_edge) => cur_edge.nonce < edge.nonce,
                        None => true,
                    }
                });
```

**Review by mfornet — APPROVED** [review]

File: chain/network/src/peer_manager.rs [hunk 2]
```diff
@@ -1450,12 +1450,24 @@ impl Handler<NetworkRequests> for PeerManagerActor {
...
 
                 self.edge_verifier_pool.send(EdgeList(edges.clone()))
-                    .into_actor(self)
-                    .then(move |response, act, ctx| {
-                        match response {
-                            Ok(false) => act.try_ban_peer(ctx, &peer_id, ReasonForBan::InvalidEdge),
+                            .into_actor(self)
```
Comment (bowenwang1996) [inline chain/network/src/peer_manager.rs:1467]: why?

**Review by bowenwang1996 — COMMENTED** [review]
Please try avoiding unnecessary reformatting. Also I'd like to mention that this is related to #3621

File: chain/network/src/peer_manager.rs (hunk 2 above)
Comment (pmnoxx) [reply chain/network/src/peer_manager.rs]: fixed, Intellij Rust sometimes automatically reformats code wrongly

**Review by bowenwang1996 — APPROVED** [review]

**Pull request merged**